scripts/rebuild-endpoints.py

```diff
@@ -39,240 +30,89 @@
-                partition)
-            for region_name in region_names:
-                endpoints[region_name] = self._resolver.get_hostname(
-                    service_name, region_name)
-        return endpoints
-
-    def _deal_with_special_cases(self, static_endpoints):
-        # I'm not sure why we do this, but cloudsearchdomain endpoints
-        # use the exact same set of endpoints as cloudsearch.
-        if 'cloudsearch' in static_endpoints:
-            static_endpoints['cloudsearchdomain'] = static_endpoints['cloudsearch']
+def filter_services(endpoints):
+    for partition in endpoints['partitions']:
+        services = list(partition['services'].keys())
+        for service in services:
+            if service not in SUPPORTED_SERVICES:
+                del partition['services'][service]
 
 
 def main():
     parser = argparse.ArgumentParser()
-    parser.add_argument('--overwrite', action='store_true')
-    parser.add_argument('--endpoints-file',
-                        help=('Path to endpoints.json.  If this argument '
-                              'is not given, then the endpoints.json file '
-                              'bundled with botocore will be used.'))
+    parser.add_argument(
+        '--endpoints-file',
+        help=('Path to endpoints.json.  If this argument is not given, '
+              'then the endpoints.json file bundled with botocore will be '
+              'used.'))
+
+    # Since the built in endpoints are no longer in the legacy format,
+    # overwrite should not be allowed with legacy format output.
+    exclusive = parser.add_mutually_exclusive_group()
+    exclusive.add_argument(
+        '--overwrite', action='store_true',
+        help=('Overwrite the endpoints file built into boto2. This is not '
+              'compatible with the legacy format.'))
+    exclusive.add_argument(
+        '--legacy-format', action='store_true',
+        help=('Generate the endpoints in the legacy format, suitable for use '
+              'as custom endpoints.'))
     args = parser.parse_args()
-    known_services_in_existing_endpoints = _load_endpoint_services(
-        EXISTING_ENDPOINTS_FILE)
-    session = botocore.session.get_session()
-    if args.endpoints_file:
-        with open(args.endpoints_file) as f:
-            endpoint_data = json.load(f)
-    else:
-        endpoint_data = session.get_data('endpoints')
-    resolver = EndpointResolver(endpoint_data)
-    strict_resolver = StrictEndpointResolver(resolver, endpoint_data)
-    builder = StaticEndpointBuilder(strict_resolver)
-    static_endpoints = builder.build_static_endpoints(
-        known_services_in_existing_endpoints)
+    endpoints_data = load_endpoints(args.endpoints_file)
+    filter_services(endpoints_data)
+
+    if args.legacy_format:
+        builder = StaticEndpointBuilder(BotoEndpointResolver(endpoints_data))
+        endpoints_data = builder.build_static_endpoints(SUPPORTED_SERVICES)
```

Comment:
I think we want to remove the ``SUPPORTED_SERVICES`` as we already filter it before: https://github.com/boto/boto/blob/endpoints-v2/boto/endpoints.py#L199 and the docstring says it needs to be boto2 naming scheme.